tests/MongoDB.Driver.Core.Tests/Core/Clusters/ClusterTests.cs

```diff
@@ -481,6 +481,65 @@ public void DescriptionChanged_should_be_raised_when_the_description_changes()
             _capturedEvents.Any().Should().BeFalse();
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void SelectServer_should_call_custom_selector(
+            [Values(true, false)] bool doNotFilterInitialServers,
+            [Values(true, false)] bool async)
+        {
+            int numberOfCustomServerSelectorCalls = 0;
+            var customServerSelector = new DelegateServerSelector((c, s) =>
+            {
+                numberOfCustomServerSelectorCalls++;
+                var highestPortServer = s.OrderByDescending(x => ((DnsEndPoint)x.EndPoint).Port).FirstOrDefault();
+
+                return highestPortServer != null ? new[] {highestPortServer} : Enumerable.Empty<ServerDescription>();
+            });
+
+            var settings = _settings.With(
+                postServerSelector: customServerSelector,
+                localThreshold: TimeSpan.FromSeconds(3));
```

Comment:
You can remove `localThreshold`, it's not necessary.

Reply:
Done.